cmd/kubeadm/app/util/config/common.go

```diff
@@ -144,3 +146,32 @@ func LowercaseSANs(sans []string) {
 		}
 	}
 }
+
+// VerifyAPIServerBindAddress can be used to verify if a bind address for the API Server is 0.0.0.0,
+// in which case this address is not valid and should not be used.
+func VerifyAPIServerBindAddress(address string) error {
+	ip := net.ParseIP(address)
+	if ip == nil {
+		return fmt.Errorf("cannot parse IP address: %s", address)
+	}
+	if !ip.IsGlobalUnicast() {
+		return fmt.Errorf("cannot use %q as the bind address for the API Server", address)
+	}
+	return nil
+}
+
+// chooseAPIServerBindAddress is a wrapper for netutil.ChooseBindAddress that also handles
+// the case where no default routes were found and an IP for the API server could not be obatained.
+func chooseAPIServerBindAddress(bindAddress net.IP, valueToUpdate *string) (net.IP, error) {
+	ip, err := netutil.ChooseBindAddress(bindAddress)
+	if err != nil {
+		if netutil.IsNoRoutesError(err) {
+			glog.Warningf("WARNING: could not obtain a bind address for the API Server: %v; using: %s", err, constants.DefaultAPIServerBindAddress)
+			*valueToUpdate = constants.DefaultAPIServerBindAddress
+			return ip, nil
+		}
+		return nil, err
+	}
+	*valueToUpdate = ip.String()
```

Comment:
The semantics are weird of this signature, you already return the ip where any consumer can call ip.String()

Reply:
I'd either call that externally or return the string.

Reply:
i will update this.